tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -165,3 +167,253 @@ def test_upgrade_brokers(self, from_version, to_version):
         node = self.driver.node
         node.account.ssh("grep ALL-RECORDS-DELIVERED %s" % self.driver.STDOUT_FILE, allow_fail=False)
         self.processor1.node.account.ssh_capture("grep SMOKE-TEST-CLIENT-CLOSED %s" % self.processor1.STDOUT_FILE, allow_fail=False)
+
+    @parametrize(old_version=str(LATEST_0_10_1), new_version=str(LATEST_0_10_2))
+    @parametrize(old_version=str(LATEST_0_10_1), new_version=str(LATEST_0_11_0))
+    @parametrize(old_version=str(LATEST_0_10_1), new_version=str(LATEST_1_0))
+    @parametrize(old_version=str(LATEST_0_10_1), new_version=str(DEV_VERSION))
+    @parametrize(old_version=str(LATEST_0_10_2), new_version=str(LATEST_0_11_0))
+    @parametrize(old_version=str(LATEST_0_10_2), new_version=str(LATEST_1_0))
+    @parametrize(old_version=str(LATEST_0_10_2), new_version=str(DEV_VERSION))
+    @parametrize(old_version=str(LATEST_0_11_0), new_version=str(LATEST_1_0))
+    @parametrize(old_version=str(LATEST_0_11_0), new_version=str(DEV_VERSION))
+    @parametrize(old_version=str(LATEST_1_0), new_version=str(DEV_VERSION))
+    def test_simple_upgrade(self, old_version, new_version):
```

Comment:
Similar to my other PR's comment, I'd suggest we update this test to get rid of using StreamsSmokeTestDriverService and use StreamsUpgradeTestJobRunnerService.

Reply:
Can you elaborate? `StreamsSmokeTestDriverService` is used the generate the input data in this test and there is no alternative implementation, and `StreamsUpgradeTestJobRunnerService` is used already.